12-watchtower-API.md

```diff
@@ -22,227 +22,320 @@ The scope of this bolt does not include:
 
-## Connection establishment
-Connections between the client and the server can be long-lived or restarted for every single appointment.
+Therefore, we are assuming that the client and server are connected and that the client have learnt what Quality of Service (`qos`) the tower is offering. Moreover, we asume that the client has an authentication token `aut_token` to prove he's entitled to use the service if required.
 
-		+-------+                      +-------+
-		|   A   |--(1)--- wt_init ---->|   B   |
-		|       |<-(2)--- wt_init -----|       |
-		+-------+                      +-------+
+## WatchTower services
+
+### Basic Service
+The customer can hire the WatchTower to watch for breaches on the blockchain and relay a justice transaction on their behalf. The customer receives an acknowledgement when the WatchTower has accepted the job, but the hiring protocol does guarantee the transaction inclusion.
+
+### Quality of Service
+Quality of Service (`qos`) builds on top of the basic service provided by a tower and it's optionally provided. Different kinds of QoS can be offered by the tower.
+
+For now we are defining a single type of `qos`: `accountability`.
+
+#### `accountability`
+
+A WatchTower provides a signed receipt to the customer. This is considered reputational accountability as the customer has publicly verifiable cryptographic evidence the WatchTower was hired. The receipt can be used to prove the WatchTower did not relay the justice transaction on their behalf and/or request a refund.
+
+## Sending and receiving appointments
+
+Once the client is aware of the services provided by the server, the former can start sending appointments to the latter.
+
+		+-------+                                    +-------+
+		|   A   |--(1)---      appointment      ---->|   B   |
+		|       |<-(2)---   accepted/rejected   -----|       |
+		+-------+                                    +-------+
 		
 		- where node A is 'client' and node B is 'server'
 
-### The `wt_init` message
+### The `appointment` message
 
-This messsage contains the information about a node and the type of appointments he is willing to create / accept.
+This message contains all the information regarding the appointment that the client wants to arrange with the server.
 
-1. type: 128 (`wt_init`)
+1. type: ? (`appointment`)
 2. data:
-   * [`u16`:`client_server`]
-   * [`u16`:`aclen`]
-   * [`aclen*bytes`:`accepted_ciphers`]
-   * [`u16`:`modlen`]
-   * [`modlen*bytes`:`modes`]
-   * [`u16`:`qoslen`]
-	* [`qoslen*bytes`:`qos`]	
+   * [`16*byte`:`locator`]
+   * [`u64 `:`start_block`]
+   * [`u64 `:`end_block`]
+   * [`u16`: `encrypted_blob_len`
+   * [`encrypted_blob_len*byte`:`encrypted_blob`]
+   * [`u16`:`cipher`]
+   * [`u16`: `auth_token_len`]
+   * [`auth_token_len*bytes`: `auth_token`]
+   * [`u16`: `qos_len`]
+   * [`qos_len*byte`: `qos_data`]
+
+#### Requirements
 
-`client_server` defines whether the sender is a client or a server.
+The sending node:
```

Comment:
Why say `sending node`, why not client? And `server` in the other cases.

I get that between lightning nodes themselves there's no concept of client or server - since all lightning nodes are running the same software, but in this protocol there is a difference. It might even be a bit confusing to call them nodes, since nodes infers a graph topology/relationship whereas in this case the topology is the simple client/sever one.

Reply:
Agreed. At some point there where messages that could have been sent by any of the parties and followed the same format. Not anymore.